### PR TITLE
Make `validate_call` return `ValidateCallWrapper`

### PR DESCRIPTION
**Pull request by kc0506**
## Change Summary

This is sort of the revert of #8268. The reason to switch back to `ValidateCallWrapper` is because I found that [supporting `Self` in `validate_call`](https://github.com/kc0506/pydantic/tree/support-validate-call-self) (WIP) can be much simplified with descriptors, since we now have access to `objtype` from `__get__`. 

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by kc0506** [comment]
@Viicos @sydney-runkle Could you give some thoughts on this? 

(Currently tests failed below Python 3.11 because `inspect.markcoroutinefunction` is added in Python 3.12)

**Comment by sydney-runkle** [comment]
Hey @kc0506,

We've really been appreciating your work here. Before you start work on a PR, could you please open an issue so we could discuss the plan? We've seen a lot of churn in the `validate_call` space that leads to a hard to follow blame + higher likelihood of introducing bugs in a minor version.

Thanks again for your work - we're looking forward to future collaboration!

**Comment by kc0506** [comment]
No problem. I will note that in the future. Should I open a new issue to discuss this PR then?

**Comment by Viicos** [comment]
We are going to close this for now, as you mentioned it is difficult to properly support it and there isn't enough upvotes on the issue to be considered worthwhile. Thanks again for taking a stab at this, we can revisit if we ever want to implement it.
